utils/db/users.ts

```diff
@@ -158,4 +158,30 @@ export const updateUser = async (userId: string, user: DbUser) => {
 			console.error('[updateUser] Update user failed', { userId, user }, err);
 			return;
 		});
+}
+
+export const getUserEmails = async (email: string): Promise<Set<string>> => {  
+	const users: DbUser[] | undefined = await getUserByAlias(email);
+	const emails = new Set<string>();
+	if (users) {
+		for (const user in users) {
+			const dbuser = users[user];
+			if (dbuser.aliases) {
+				for (const idx in dbuser.aliases) {
+					emails.add(dbuser.aliases[idx]);
+				}
+			}
+			if (dbuser.auth_info) {
+				for (const provider in dbuser.auth_info) {
+					for (const account in dbuser.auth_info[provider]) {
+						const auth = dbuser.auth_info[provider][account];
```

Comment:
```suggestion
				for (const provider of dbuser.auth_info) {
					for (const auth of provider) {
```

Reply:
auth_info is a dictionary, for...of doesn't work